ipn/ipnlocal/serve.go

```diff
@@ -141,26 +159,84 @@ func (b *LocalBackend) serveWebHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	if v := h.Proxy(); v != "" {
-		io.WriteString(w, "TODO(bradfitz): proxy")
+		// TODO(bradfitz): this is a lot of setup per HTTP request. We should
+		// build the whole http.Handler with all the muxing and child handlers
+		// only on start/config change. But this works for now (2022-11-09).
+		u, err := url.Parse(expandProxyArg(v))
+		if err != nil {
+			http.Error(w, "bad proxy config", http.StatusInternalServerError)
+			return
+		}
+		rp := httputil.NewSingleHostReverseProxy(u)
+		rp.Transport = &http.Transport{
+			DialContext: b.dialer.SystemDial,
+		}
+		rp.ServeHTTP(w, r)
 		return
 	}
 
 	http.Error(w, "empty handler", 500)
 }
 
-func (b *LocalBackend) getTLSServeCert(hi *tls.ClientHelloInfo) (*tls.Certificate, error) {
-	if hi == nil || hi.ServerName == "" {
-		return nil, errors.New("no SNI ServerName")
+// expandProxyArg returns a URL from s, where s can be of form:
+//
+// * port number ("8080")
+// * host:port ("localhost:8080")
+// * full URL ("http://localhost:8080", in which case it's returned unchanged)
+func expandProxyArg(s string) string {
+	if s == "" {
+		return ""
+	}
+	if strings.HasPrefix(s, "http://") || strings.HasPrefix(s, "https://") {
+		return s
 	}
```

Comment:
Just a thought... if we want to support a local https proxy with a (maybe self signed/invalid) cert. Weird case, but we'd do termination but the backend is https. The self signed case would require a `InsecureSkipVerify`)

Reply:
Go at least uses a protocol scheme of `https+insecure://` for that. We could do the same?